kubernetes/src/kubernetes/watch.rb

```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'json'
+
+# The Kubernetes module encapsulates the Kubernetes client for Ruby
+module Kubernetes
+  # The Watch class provides the ability to watch a specific resource for
+  # updates.
+  class Watch
+    def initialize(client)
+      @client = client
+    end
+
+    def connect(path, &_block)
+      opts = { auth_names: ['BearerToken'] }
+      request = @client.build_request('GET', path + '?watch=true', opts)
+      last = ''
+      request.on_body do |chunk|
+        last, pieces = split_lines(last, chunk)
+        pieces.each do |part|
+          yield JSON.parse(part)
+        end
+      end
+      request.run
+    end
+
+    def split_lines(last, chunk)
+      data = chunk
+      data = last + "\n" + data unless last.empty?
```

Comment:
the idea was that a single line could(?) be split between prev chunk and this one, so I think this should just be
```suggestion
data = last + chunk
```

Reply:
No, because if last itself is empty you don't want to randomly prepend a newline...  (I had that bug)

Reply:
Right but I'm saying you don't want the newline between them!
Say first chunk is cut `{"Foo": 0}\n{"ba`
and next chunk continues it `r": 1}\n{"baz`.
If you join with a newline, the 2nd json will not parse properly.


Then, prepending an empty string is harmless, you can keep the `unless last.empty?` or drop it, as you prefer.

Reply:
fixed.